hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/retransmit/KafkaRetransmissionService.java

```diff
@@ -71,50 +59,21 @@ public KafkaRetransmissionService(
         return partitionOffsetList;
     }
 
-    private SimpleConsumer createSimpleConsumer(KafkaTopicName kafkaTopicName, int partition) {
-        Integer leader = brokerStorage.readLeaderForPartition(new TopicAndPartition(kafkaTopicName.asString(), partition));
-
-        return simpleConsumerPool.get(leader);
-    }
-
-    private long getLastOffset(SimpleConsumer consumer, Topic topic, KafkaTopic kafkaTopic, int partition, long timestamp) {
-        Range<Long> offsetRange = getOffsetRange(consumer, kafkaTopic, partition);
-        return search(topic, kafkaTopic, partition, offsetRange, timestamp);
-    }
-
-    private long search(Topic topic, KafkaTopic kafkaTopic, int partition, Range<Long> offsetRange, long timestamp) {
-        OffsetSearcher searcher = new OffsetSearcher(
-                new KafkaTimestampExtractor(topic, kafkaTopic, partition, kafkaRawMessageReader, messageContentWrapper)
-        );
-        return searcher.search(offsetRange, timestamp);
-    }
-
-    private Range<Long> getOffsetRange(SimpleConsumer simpleConsumer, KafkaTopic kafkaTopic, int partition) {
-        long earliestOffset = getOffset(simpleConsumer, kafkaTopic, partition, OffsetRequest.EarliestTime());
-        long latestOffset = getOffset(simpleConsumer, kafkaTopic, partition, OffsetRequest.LatestTime());
-
-        return Range.closed(earliestOffset, latestOffset);
+    private KafkaConsumer<byte[], byte[]> createKafkaConsumer(KafkaTopic kafkaTopic, int partition) {
+        return consumerPool.get(kafkaTopic, partition);
     }
 
-    private long getOffset(SimpleConsumer simpleConsumer, KafkaTopic topicName, int partition, long whichTime) {
-        TopicAndPartition topicAndPartition = new TopicAndPartition(topicName.name().asString(), partition);
-
-        Map<TopicAndPartition, PartitionOffsetRequestInfo> requestInfo = new HashMap<>();
-        requestInfo.put(topicAndPartition, new PartitionOffsetRequestInfo(whichTime, 1));
-
-        kafka.javaapi.OffsetRequest request =
-                new kafka.javaapi.OffsetRequest(requestInfo, OffsetRequest.CurrentVersion(), "KafkaRetransmissionService" + topicName.name().asString());
-        OffsetResponse response = simpleConsumer.getOffsetsBefore(request);
-
-        return readOffsetFromResponse(response, topicAndPartition);
+    private long findClosestOffsetJustBeforeTimestamp(KafkaConsumer<byte[], byte[]> consumer, KafkaTopic kafkaTopic, int partition, long timestamp) {
```

Comment:
Do we need `Just` in the name ? ;D 
(JUST kidding)

Reply:
It ```just``` puts emphasis on closest word :D